guides-and-standards/guide-choosing-software-dependencies.md

```diff
@@ -0,0 +1,59 @@
+# Selecting software dependencies for workflows and software
+
+We often expect that a computational analysis we write at Arcadia will either be used again by another Arcadian or automated for high throughput studies. When this is the case, the software quality used in those analyses becomes more important than if the analysis is a one-off task used to generate a specific biological insight.
+
+## Deciding when your task meets the re-use bar
+
+**When working on a software tool or workflow you anticipate that you or another Arcadian will use multiple times, you need to be careful about which tools you include**. This is _not_ the same as a workflow that documents the steps taken in an analysis project and is often bespoke and specific to a given task. Examples of existing tools or workflows where we should be careful about the quality of tools we use include:
+
+- nf-core-style Nextflow workflows like [NovelTree](https://github.com/Arcadia-Science/noveltree), [PreHGT](https://github.com/Arcadia-Science/prehgt), and [reads2genome](https://github.com/Arcadia-Science/reads2genome)
+- Snakemake workflows written as re-usable tools like [ProteinCartography](https://github.com/Arcadia-Science/ProteinCartography/)
+- Python packages like [plm-utils](https://github.com/Arcadia-Science/2024-plm-utils).
+- Tools designed to analyze experimental data in high throughput like [arcadia-phenotypeomat-protocol]([https://github.com/Arcadia-Science/arcadia-phenotypeomat-protocol)
+
+## Things to look for when deciding to include a tool as a dependency
+
+A dependency is any software tool that is required to make your tool work. For example, BLAST is a dependency of ProteinCartography, and Nextflow and OrthoFinder are dependencies of NovelTree. Below we include some signals for assessing whether a software tool is high quality.
+
+### Critical — we can’t use the tool if these criteria are not met
+
+- **The tool has a license and the license is permissible for commercial use.** Common licenses like BSD2, BSD3, GPL3, and MIT can all be used. Some other licenses can be used as well.
+  - Hint: you can search for the word "commercial" in the license. This will often take you to text about whether the software can be used commercially. If it can’t, it will usually say something like "for non-commercial use only."
+- **The tool can be run on the command line without human input.** A Graphical User Interface (GUI) (including web applications like Streamlit) is not required to run the tool. The tool isn’t encoded in notebooks.
```

Comment:
Maybe expand on the notebook requirement? We've used ColabFold for example for different projects and that's technically a tool encoded in notebooks if you want to run it on Google Colab

Reply:
what do you think I should add here? I think ColabFold would be a very bad thing to include as a dependency for a high-throughput or software tool because of the requirement to use in in Google colab. It's really good to test an analysis, but if we want to scale up the thing, we'll have to come up with a different solution if ColabFold is the answer.

Reply:
ah sorry, I think my brain shifted from how to implement a tool in a high-throughput workflow vs just being able to use it at all. Here, I'm not sure what to recommend if you want to create something reusable because, especially for accessing GPUs, Colab would be an easier way to share/reuse/launch than somebody setting up an instance for GPUs. I think there's sometimes an intermediate point between making something for a one-off and making a fully reproducible workflow/tool since we sometimes aren't sure if the effort to make it fully high-throughput would be worthwhile. I'm sort of thinking about the PPI analysis I did, it would be easy for somebody to set that up and get going themselves, and then we assess over time if we make a different, more high-throughput solution. This might beyond the scope of this guide, but those are the thoughts I have.

Reply:
I think Colabfold actually is fine in the critera Taylor described. The Colabfold notebook is the equivalent of a convenient streamlit app. If you want to you can install Colabfold via conda and run it via Snakemake or Nextflow or even locally. And they probably even offer Docker containers, but I know it's pip/conda installable.